### PR TITLE
Chore: fix all warns in `cargo check --tests --workspace`

### PR DESCRIPTION
**Pull request by kantai**
This PR fixes all the warnings generated by current rust stable in `develop` when run with `cargo check --workspace --tests` (previously, warnings were only addressed in `cargo check`)

### CODE REVIEW

File: src/net/http.rs [hunk 1]
```diff
@@ -4895,9 +4895,10 @@ mod test {
             let res = HttpRequestPreamble::consensus_deserialize(&mut data.as_bytes());
             test_debug!("Expect '{}'", errstr);
             let expected_errstr = format!("{:?}", &res);
```
Comment (pavitthrap) [inline src/net/http.rs:4897]: could simplify to `assert!(res.is_err(), "{:?}", &res);` and delete previous line

File: src/net/http.rs [hunk 2]
```diff
@@ -4948,9 +4949,10 @@ mod test {
             let sres = StacksHttpPreamble::consensus_deserialize(&mut data.as_bytes());
             let expected_serrstr = format!("{:?}", &sres);
             test_debug!("Expect '{}'", errstr);
-            assert!(sres.is_err(), expected_serrstr);
+            assert!(sres.is_err(), "{}", expected_serrstr);
```
Comment (pavitthrap) [inline src/net/http.rs:4952]: can simplify the assert here as well

File: src/net/http.rs [hunk 3]
```diff
@@ -5213,9 +5215,10 @@ mod test {
             let sres = StacksHttpPreamble::consensus_deserialize(&mut data.as_bytes());
             let expected_serrstr = format!("{:?}", &sres);
             test_debug!("Expect '{}', got: {:?}", errstr, &sres);
-            assert!(sres.is_err(), expected_serrstr);
+            assert!(sres.is_err(), "{}", expected_serrstr);
```
Comment (pavitthrap) [inline src/net/http.rs:5218]: can simplify here

File: src/net/http.rs [hunk 4]
```diff
@@ -5464,12 +5467,13 @@ mod test {
             let e = http.read_payload(&preamble, &bad_content_length.as_bytes()[offset..]);
             let estr = format!("{:?}", &e);
 
-            assert!(e.is_err(), estr);
+            assert!(e.is_err(), "{}", estr);
```
Comment (pavitthrap) [inline src/net/http.rs:5470]: can simplify here

File: src/util/retry.rs [hunk 5]
```diff
@@ -169,11 +169,12 @@ mod test {
         let mut tmp_buf = [0u8; 3];
         let e = retry_reader.read_exact(&mut tmp_buf);
         let e_str = format!("{:?}", &e);
-        assert!(e.is_err(), e_str);
+        assert!(e.is_err(), "{}", e_str);
```
Comment (pavitthrap) [inline src/util/retry.rs:172]: can simplify here

**Review by pavitthrap — APPROVED** [review]
Just left some minor comments about removing some calls to `format!` - but LGTM!

**Comment by gregorycoppola** [comment]
Should we get all these changes in `next` as well?

Is the easiest way to do that two PR's?

**Review by jcnelson — APPROVED** [review]

**Comment by kantai** [comment]
> Should we get all these changes in next as well?
> Is the easiest way to do that two PR's?

Yes -- and sort of. For "upstream" updates, i.e., `master -> develop` or `develop -> next`, we shouldn't really need a review. However, branch protection rules mean that we in practice need that rubber stamp. So yeah, opening a PR is probably the easiest way to achieve that.

File: src/util/retry.rs (hunk 5 above)
Comment (kantai) [reply src/util/retry.rs]: Addressed in 93a7c393069ca4420433e03338a6534bee19dc8f

File: src/net/http.rs (hunk 4 above)
Comment (kantai) [reply src/net/http.rs]: Addressed in 93a7c393069ca4420433e03338a6534bee19dc8f

File: src/net/http.rs (hunk 2 above)
Comment (kantai) [reply src/net/http.rs]: Addressed in 93a7c393069ca4420433e03338a6534bee19dc8f

File: src/net/http.rs (hunk 3 above)
Comment (kantai) [reply src/net/http.rs]: Addressed in 93a7c393069ca4420433e03338a6534bee19dc8f

File: src/net/http.rs (hunk 1 above)
Comment (kantai) [reply src/net/http.rs]: Addressed in 93a7c393069ca4420433e03338a6534bee19dc8f

**Pull request merged**